constraint-solver/src/solver/base.rs

```diff
@@ -560,6 +560,10 @@ where
     }
 }
 
+/// Returns true if the expression is equivalent to `X = Y` for
```

Comment:
What does it mean for an expression to be equivalent to `X = Y`? Is it when the expression is seen as an `assert_zero` constraint?